lib/AllocateDescriptorsPass.cpp

```diff
@@ -95,20 +95,27 @@ bool clspv::AllocateDescriptorsPass::AllocateLiteralSamplerDescriptors(
 
   // Generate the function type for clspv::LiteralSamplerFunction()
   IRBuilder<> Builder(M.getContext());
-  auto *sampler_struct_ty =
-      StructType::getTypeByName(M.getContext(), "opencl.sampler_t");
-  if (!sampler_struct_ty) {
-    sampler_struct_ty = StructType::create(M.getContext(), "opencl.sampler_t");
-  }
-  // TODO: #816 remove after final switch.
   Type *sampler_ty = nullptr;
-  if (init_fn->getType()->isOpaquePointerTy()) {
-    sampler_ty = PointerType::get(M.getContext(), clspv::AddressSpace::Constant);
+  Type *sampler_data_ty = nullptr;;
+  // TODO(#1036): remove opaque struct support
+  if (init_fn->getReturnType()->isTargetExtTy()) {
+    sampler_ty = init_fn->getReturnType();
+    sampler_data_ty = init_fn->getReturnType();
```

Comment:
With the old style of types, the return type was a pointer (default opaque) whilst the data type is a struct. Now that it is a bare value the types are the same. This can be reduced to a single variable as part of #1036.